### PR TITLE
[dashboard] Add ItemsList component

### LINKED ISSUE #3731: Keep hover state and dropdown visible when more actions dropdown is expanded
### Probem to solve

The row hover state on _Workspaces_ is removed when you move mouse pointer outside the row.

This also affects _Integrations_, _Variables_, and _Teams_.

https://user-images.githubusercontent.com/120486/113421863-729a5980-93d4-11eb-843c-ccfaa9fbad32.mov

### Proposal

Keep hover state and dropdown visible when more actions dropdown is expanded. Only change hover state and dropdown visibility when the user clicks outside the row area.

### LINKED ISSUE #4363: Keep the more actions button (three dots) always visible
### Problem to solve

The more actions button (three dots) can be hard a) to discover as a user and b) to describe in the documentation as 
a user interaction. See [relevant discussion](https://gitpod.slack.com/archives/C01KGM9BUNS/p1622537621030100) (internal). /cc @JanKoehnlein 

This affects the following pages which contain list entries with a disappearring more actions button:
1. **Workspaces**
2. **Variables**
3. **Integrations**
4. **Teams**

Alternatively, surfacing more action on hover could limit or remove the need to use the more actions button, see https://github.com/gitpod-io/gitpod/issues/4213.

### Risk

It'd be worth checking if the interaction can cause any confusion due to this issue (see https://github.com/gitpod-io/gitpod/issues/3731).

### Proposal

Let's keep the more actions button (three dots) always visible on all list items.

| BEFORE | AFTER |
|-|-|
| image | image |

The fix should be relatively simple by repeating the following on every instance:

```diff
diff --git a/components/dashboard/src/settings/Integrations.tsx b/components/dashboard/src/settings/Integrations.tsx
index f4fe2be3..8a5bea48 100644
--- a/components/dashboard/src/settings/Integrations.tsx
+++ b/components/dashboard/src/settings/Integrations.tsx
@@ -283,7 +283,7 @@ function GitProviders() {
         <h2>Manage permissions for git providers.</h2>
         <div className="flex flex-col pt-6 space-y-2">
             {authProviders && authProviders.map(ap => (
-                <div key={"ap-" + ap.authProviderId} className="flex-grow flex flex-row hover:bg-gray-100 dark:hover:bg-gray-800 rounded-xl h-16 w-full transition ease-in-out group">
+                <div key={"ap-" + ap.authProviderId} className="flex-grow flex flex-row hover:bg-gray-100 dark:hover:bg-gray-800 rounded-xl h-16 w-full transition ease-in-out">
                     <div className="px-4 self-center w-1/12">
                         <div className={"rounded-full w-3 h-3 text-sm align-middle " + (isConnected(ap.authProviderId) ? "bg-green-500" : "bg-gray-400")}>
                             &nbsp;
@@ -417,7 +417,7 @@ function GitIntegrations() {
                     <div className="p-0 my-auto flex flex-col w-7/12">
                         <span className="my-auto truncate text-gray-500 overflow-ellipsis">{ap.host}</span>
                     </div>
-                    <div className="my-auto flex w-1/12 mr-4 opacity-0 group-hover:opacity-100 justify-end">
+                    <div className="my-auto flex w-1/12 mr-4 justify-end">
                         <div className="self-center hover:bg-gray-200 dark:dark:bg-gray-800 rounded-md cursor-pointer w-8">
                             <ContextMenu menuEntries={gitProviderMenu(ap)}>
                                 <svg className="w-8 h-8 p-1 text-gray-600 dark:text-gray-300" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><title>Actions</title><g fill="currentColor" transform="rotate(90 12 12)"><circle cx="1" cy="1" r="2" transform="translate(5 11)" /><circle cx="1" cy="1" r="2" transform="translate(11 11)" /><circle cx="1" cy="1" r="2" transform="translate(17 11)" /></g></svg>
```

### PR DESCRIPTION
**Pull request by corneliusludmann**
This PR adds an `ItemsList` component that replaces the divs at the following places:
- Worksapaces List
- Settings → Environment Varialbes
- Settings → Integrations
- Settings → Teams

This makes these nearly identical HTML structures more maintainable and consistent.

This also fixes #4363 and fixes #3731.

/cc @gtsiolis

TODO:
- [x] Test Teams page (How can I test this without payment endpoint?)
- [x] Resolve conflict with #4416 (probably add the changes here instead).

### CODE REVIEW

File: components/dashboard/src/workspaces/Workspaces.tsx [hunk 1]
```diff
@@ -104,33 +105,33 @@ export default class Workspaces extends React.Component<WorkspacesProps, Workspa
...
                         {
                             wsModel.active || wsModel.searchTerm ? null :
-                                <div className="whitespace-nowrap flex space-x-2 py-6 px-6 w-full justify-between bg-gitpod-kumquat-light rounded-xl">
-                                    <div className="pr-3 self-center w-6">
-                                        <img src={exclamation} />
-                                    </div>
-                                    <div className="flex-1 flex flex-col overflow-x-auto">
+                                <Item className="w-full bg-gitpod-kumquat-light py-6 px-6">
+                                    <ItemFieldIcon>
+                                        <img src={exclamation} alt="Exclamation Mark" className="m-auto" />
```
Comment (gtsiolis) [inline components/dashboard/src/workspaces/Workspaces.tsx:121]: **praise**: Thanks for adding the `alt` text here! 🏅

Cross-posting https://github.com/gitpod-io/gitpod/issues/3508 for visibility and future reference.

File: components/dashboard/src/workspaces/Workspaces.tsx [hunk 2]
```diff
@@ -104,33 +105,33 @@ export default class Workspaces extends React.Component<WorkspacesProps, Workspa
...
-                            <div className="w-8"></div>
-                        </div>
+                    <ItemsList className="lg:px-28 px-10">
+                        <Item header={true} className="px-6">
+                            <ItemFieldIcon />
+                            <ItemField className="w-3/12">Name</ItemField>
+                            <ItemField className="w-4/12">Context</ItemField>
+                            <ItemField className="w-2/12">Pending Changes</ItemField>
+                            <ItemField className="w-2/12">Last Start</ItemField>
+                            <ItemFieldContextMenu />
```
Comment (gtsiolis) [inline components/dashboard/src/workspaces/Workspaces.tsx:115]: **praise**: Love at first sight! ❤️

**Review by gtsiolis — APPROVED** [review]
Whoa, thanks for adding this @corneliusludmann! Works like a charm. 🔮 

FWIW, the current teams settings section will probably become obsolete and be removed once we have the new Teams & Projects in place.

**Comment by corneliusludmann** [comment]
> FWIW, the current teams settings section will probably become obsolete and be removed once we have the new Teams & Projects in place.

OK, good to know. Then, I think it's a good idea to revert the changes in the Teams page for now and merge this afterwards as is.


Thanks for your review! :+1:

**Pull request merged**